src/main/java/net/dv8tion/jda/api/utils/AttachedFile.java

```diff
@@ -0,0 +1,291 @@
+     * <br>This is used to upload data to discord for various purposes.
+     *
+     * <p>This opens a {@link FileInputStream}, which will be closed on consumption by the request.
+     * You can use {@link FileUpload#close()} to close the stream manually.
+     *
+     * @param  file
+     *         The {@link File} to upload
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided
+     * @throws UncheckedIOException
+     *         If an IOException is thrown while opening the file
+     *
+     * @return {@link FileUpload}
+     *
+     * @see    java.io.FileInputStream FileInputStream
+     * @see    #fromData(File, String)
+     */
+    @Nonnull
+    static FileUpload fromData(@Nonnull File file)
+    {
+        return FileUpload.fromData(file);
+    }
+
+    /**
+     * Create a new {@link FileUpload} for a local file.
+     * <br>This is used to upload data to discord for various purposes.
+     *
+     * <p>This opens the path using {@link Files#newInputStream(Path, OpenOption...)}, which will be closed on consumption by the request.
+     * You can use {@link FileUpload#close()} to close the stream manually.
+     *
+     * @param  path
+     *         The {@link Path} of the file to upload
+     * @param  name
+     *         The representative name to use for the file
+     * @param  options
+     *         The {@link OpenOption OpenOptions} specifying how the file is opened
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided or the name is empty
+     * @throws UncheckedIOException
+     *         If an IOException is thrown while opening the file
+     *
+     * @return {@link FileUpload}
+     */
+    @Nonnull
+    static FileUpload fromData(@Nonnull Path path, @Nonnull String name, @Nonnull OpenOption... options)
+    {
+        return FileUpload.fromData(path, name, options);
+    }
+
+    /**
+     * Create a new {@link FileUpload} for a local file.
+     * <br>This is used to upload data to discord for various purposes.
+     * Uses {@link Path#getFileName()} to specify the name of the file.
+     *
+     * <p>This opens the path using {@link Files#newInputStream(Path, OpenOption...)}, which will be closed on consumption by the request.
+     * You can use {@link FileUpload#close()} to close the stream manually.
+     *
+     * @param  path
+     *         The {@link Path} of the file to upload
+     * @param  options
+     *         The {@link OpenOption OpenOptions} specifying how the file is opened
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided
+     * @throws UncheckedIOException
+     *         If an IOException is thrown while opening the file
+     *
+     * @return {@link FileUpload}
+     */
+    @Nonnull
+    static FileUpload fromData(@Nonnull Path path, @Nonnull OpenOption... options)
+    {
+        return FileUpload.fromData(path, path.getFileName().toString(), options);
```

Comment:
`getFileName` can return `null`.

Side note, what happens here for a Path like `/my/cool/path/`? does it return `path` as the `fileName`?

Reply:
It strips the trailing `/` and returns `path`